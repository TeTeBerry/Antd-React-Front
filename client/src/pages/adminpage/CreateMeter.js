import React, { Component } from "react";
import { Button, message } from "antd";
import axios from "axios";
import CollectionCreateForm from "./CreateMeterForm";
const token = localStorage.getItem("id_token");
const headers = {
  auth: token
};
class CreateMeter extends Component {
  state = {
    visible: false
  };

  error = () => {
    message.error("Add Fail");
  };

  showModal = () => {
    this.setState({ visible: true });
  };

  handleCancel = () => {
    this.setState({ visible: false });
  };

  createSuccess = () => {
    message.success("Create success!");
  };

  handleCreate = () => {
    const { form } = this.formRef.props;
    const formFields = form.getFieldsValue();
    console.log(formFields.meterDesc);
    form.validateFields((err, values) => {
      if (err) {
        return;
      }
      console.log("Received values of form: ", values);
      form.resetFields();
      this.setState({ visible: false });
    });
    const member = {
      meterDesc: formFields.meterDesc,
      meterName: formFields.meterName,
      memberName: formFields.memberName,
      room: formFields.room,
      memberContact: formFields.memberContact
    };
    axios
      .post("http://localhost:8080/iot/admin/addMeter", member, {
        headers
      })
      .then(member => {
        this.props.coolName(member);
        this.createSuccess();
      })
      .catch(error => {
        console.log(error);
        this.error();
      });
  };

  saveFormRef = formRef => {
    this.formRef = formRef;
  };

  render() {
    return (
      <div>
        <Button type="primary" onClick={this.showModal}>
          Create Meter
        </Button>
        <CollectionCreateForm
          wrappedComponentRef={this.saveFormRef}
          visible={this.state.visible}
          onCancel={this.handleCancel}
          handleSubmit={this.handleCreate}
        />
      </div>
    );
  }
}

export default CreateMeter;
