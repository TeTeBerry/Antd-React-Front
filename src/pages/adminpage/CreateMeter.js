import React, { Component } from "react";
import { Button, message } from "antd";
import axios from "axios";
import CollectionCreateForm from "./CreateMeterForm";
import AuthService from "../auth/AuthService";
import q from "querystring";

class CreateMeter extends Component {
  Auth = new AuthService();
  state = {
    visible: false
  };

  error = () => {
    message.error("Add Fail");
  };

  showModal = () => {
    this.setState({ visible: true });
  };

  handleCancel = () => {
    this.setState({ visible: false });
  };

  createSuccess = () => {
    message.success("Create success!");
  };

  handleCreate = () => {
    const { form } = this.formRef.props;
    const formFields = form.getFieldsValue();
    const member = {
      meterDesc: formFields.meterDesc,
      meterName: formFields.meterName,
      name: formFields.name,
      room: formFields.room,
      contact: formFields.contact
    };
    const user_id = q.stringify({
      user_id: this.Auth.getUserId()
    });
    form.validateFields((err, values) => {
      if (err) {
        return;
      }
      console.log("Received values of form: ", values);
      this.setState({ visible: false });
      form.resetFields();
    });

    const token = this.Auth.getToken();

    axios
      .post(`/iot/meter/addMeter/?${user_id}`, member, {
        headers: {
          token: token
        }
      })
      .then(member => {
        if (member.data.code !== 200) {
          return this.error();
        }

        this.props.coolName(member);
        this.createSuccess();
      })
      .catch(error => {
        console.log(error);
        this.error();
        console.log(token);
      });
  };

  saveFormRef = formRef => {
    this.formRef = formRef;
  };

  render() {
    return (
      <div>
        <Button
          type="primary"
          onClick={this.showModal}
          style={{ marginTop: 6 }}
        >
          Create Meter
        </Button>
        <CollectionCreateForm
          wrappedComponentRef={this.saveFormRef}
          visible={this.state.visible}
          onCancel={this.handleCancel}
          handleSubmit={this.handleCreate}
        />
      </div>
    );
  }
}

export default CreateMeter;
