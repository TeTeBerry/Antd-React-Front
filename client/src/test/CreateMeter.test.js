import React from "react";
import setTest from "../setupTests";
import { shallow } from "enzyme";
import CreateMeter from "../pages/adminpage/CreateMeter";

describe("Test case for CreateMeter", () => {
  it("get member list ", () => {
    const wrapper = shallow(<CreateMeter />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
