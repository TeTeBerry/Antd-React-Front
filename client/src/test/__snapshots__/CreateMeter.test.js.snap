// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for CreateMeter get member list  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateMeter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Create Meter
        </Button>,
        <Form(CreateMember)
          handleSubmit={[Function]}
          onCancel={[Function]}
          visible={false}
          wrappedComponentRef={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "children": "Create Meter",
          "ghost": false,
          "htmlType": "button",
          "loading": false,
          "onClick": [Function],
          "type": "primary",
        },
        "ref": null,
        "rendered": "Create Meter",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
          "onCancel": [Function],
          "visible": false,
          "wrappedComponentRef": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            type="primary"
          >
            Create Meter
          </Button>,
          <Form(CreateMember)
            handleSubmit={[Function]}
            onCancel={[Function]}
            visible={false}
            wrappedComponentRef={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Create Meter",
            "ghost": false,
            "htmlType": "button",
            "loading": false,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": "Create Meter",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
            "onCancel": [Function],
            "visible": false,
            "wrappedComponentRef": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
